aten/src/ATen/native/mps/operations/Attention.mm

```diff
@@ -19,6 +19,17 @@
 namespace at {
 namespace native {
 
+// expand potential 3d to 4d tensor
+static inline std::tuple<Tensor, bool> ensure_4d(const Tensor& x) {
+    bool squeezed = false;
+    auto t = x;
+    if (x.dim() == 3) {
+        t = x.unsqueeze(0);
+        squeezed = true;
+    }
+    return {t, squeezed};
```

Comment:
```suggestion
    return {std::move(t), squeezed};
```
Pretty sure, t is copied from x in this case on line 25. We should avoid an extra unecessary increment here.